whelktool/scripts/cleanups/2019/10/lxl-2741-remove-properties-from-genreform/main.groovy

```diff
@@ -0,0 +1,33 @@
+/*
+See LXL-2741 & LXL-2395 for more info.
+*/
+
+PrintWriter failedIDs = getReportWriter("failed-to-delete-bibIDs")
+PrintWriter scheduledForChange = getReportWriter("scheduledForChange")
+
+selectBySqlWhere("collection = 'auth' AND data#>>'{@graph,1,@type}' = 'GenreForm'", silent: false) { documentItem ->
+    def record = documentItem.doc.data.get('@graph')[0]
+
+    removeFieldFromRecord(record, "descriptionConventions", scheduledForChange)
+    removeFieldFromRecord(record, "marc:catalogingSource", scheduledForChange)
+    removeFieldFromRecord(record, "marc:kindOfRecord", scheduledForChange)
+    removeFieldFromRecord(record, "marc:level", scheduledForChange)
+    removeFieldFromRecord(record, "marc:numberedSeries", scheduledForChange)
+    removeFieldFromRecord(record, "marc:personalName", scheduledForChange)
+    removeFieldFromRecord(record, "marc:reference", scheduledForChange)
+    removeFieldFromRecord(record, "marc:romanization", scheduledForChange)
+    removeFieldFromRecord(record, "marc:transcribingAgency", scheduledForChange)
+    removeFieldFromRecord(record, "marc:typeOfSeries", scheduledForChange)
+
```

Comment:
Is it worth to check if anything has been changed before saving @niklasl? Maybe add such a check to avoid overhead.

Reply:
Yes, it is. The `scheduleForUpdate` part is designed to be called whenever a change is performed (it is only saved once, regardless of how many times `scheduleForUpdate` is called). We do not want to save unmodified records.